CMakeLists.txt

```diff
@@ -85,7 +85,10 @@ if(BUILD_TESTING)
   find_package(ament_cmake_cpplint REQUIRED)
   find_package(ament_cmake_lint_cmake REQUIRED)
   find_package(ament_cmake_uncrustify REQUIRED)
-  ament_cppcheck()
+  # This forces cppcheck to consider all files in this project to be C++,
+  # including the headers which end with .h, which cppcheck would normally
+  # consider to be C instead.
+  ament_cppcheck("--language" "c++")
```

Comment:
This is not how the API is supposed to be used: https://github.com/ament/ament_lint/blob/bdfec5611b02377f568226afce2d97725fea4c5f/ament_cmake_cppcheck/cmake/ament_cppcheck.cmake#L16-L23

I would recommend to add this option explicitly to the API rather than relying on this usage to work.